File: pdu/python/hako_srv_msgs/pdu_cdr_conv_GetSimStateRequest.py
from .pdu_pytype_GetSimStateRequest import GetSimStateRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_GetSimStateRequest(writer: CdrWriter, src: GetSimStateRequest):
    writer.write_string(src.name)


def cdr_body_to_py_GetSimStateRequest(reader: CdrReader, dst: GetSimStateRequest):
    dst.name = reader.read_string()
    return dst


def py_to_cdr_GetSimStateRequest(src: GetSimStateRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_GetSimStateRequest(writer, src)
    return writer.bytes()


def cdr_to_py_GetSimStateRequest(cdr_payload) -> GetSimStateRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = GetSimStateRequest()
    return cdr_body_to_py_GetSimStateRequest(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_GetSimStateRequest.js
import { GetSimStateRequest } from './pdu_jstype_GetSimStateRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduGetSimStateRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { GetSimStateRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.name);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { GetSimStateRequest } dst
     * @returns { GetSimStateRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.name = reader.read_string();
        return dst;
    }

    /**
     * @param { GetSimStateRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { GetSimStateRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new GetSimStateRequest());
    }
}
